gfx/drivers_context/wgl_ctx.c

```diff
@@ -274,29 +274,79 @@ static void create_gl_context(HWND hwnd, bool *quit)
 
+      /* In order to support the core info "required_hw_api" field correctly, we should try to init the highest available
+       * version GL context possible. This means trying successively lower versions until it works, because GL has
+       * no facility for determining the highest possible supported version.
+       */
       if (pcreate_context)
       {
-         HGLRC context = pcreate_context(win32_hdc, NULL, attribs);
-
-         if (context)
+         int i;
+         int gl_versions[][2] = {{4, 6}, {4, 5}, {4, 4}, {4, 3}, {4, 2}, {4, 1}, {4, 0}, {3, 3}, {3, 2}, {0, 0}};
+         int gl_version_rows = ARRAY_SIZE(gl_versions);
+         int (*versions)[2];
+         int version_rows = 0;
+         HGLRC context = NULL;
+
+         versions = gl_versions;
+         version_rows = gl_version_rows;
+
+         /* try each version, starting with the highest first */
+         for (i = 0; i < version_rows; i++)
          {
-            wglMakeCurrent(NULL, NULL);
-            wglDeleteContext(win32_hrc);
-            win32_hrc = context;
-            if (!wglMakeCurrent(win32_hdc, win32_hrc))
-               *quit = true;
-         }
-         else
-            RARCH_ERR("[WGL]: Failed to create core context. Falling back to legacy context.\n");
+            if (versions[i][0] == 0 && versions[i][1] == 0)
+            {
+               /* use the actual requested version last */
+               versions[i][0] = win32_major;
+               versions[i][1] = win32_minor;
+            }
 
-         if (win32_use_hw_ctx)
-         {
-            win32_hw_hrc = pcreate_context(win32_hdc, context, attribs);
-            if (!win32_hw_hrc)
+            attribs[1] = versions[i][0];
+            attribs[3] = versions[i][1];
+
+            context = pcreate_context(win32_hdc, NULL, attribs);
+
+            if (context)
             {
-               RARCH_ERR("[WGL]: Failed to create shared context.\n");
-               *quit = true;
+               wglMakeCurrent(NULL, NULL);
+               wglDeleteContext(win32_hrc);
+               win32_hrc = context;
+
+               if (!wglMakeCurrent(win32_hdc, win32_hrc))
+               {
+                  *quit = true;
+                  break;
+               }
+            }
+            else
+            {
+               if (versions[i][0] == win32_major && versions[i][1] == win32_minor)
+               {
+                  /* The requested version is not supported, go ahead and fail since everything else will be lower than that. */
+                  break;
+               }
+
+               continue;
+            }
+
+            if (win32_use_hw_ctx)
+            {
+               win32_hw_hrc = pcreate_context(win32_hdc, context, attribs);
+
+               if (!win32_hw_hrc)
+               {
+                  RARCH_ERR("[WGL]: Failed to create shared context.\n");
+                  *quit = true;
+                  break;
+               }
             }
+
+            break;
```

Comment:
The whole `for` loop would be more readable if the `if (win32_use_hw_ctx)` branch and the following `break` is moved inside the `if (context)` after `wglMakeCurrent`